invenio_collections/models.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-Collections is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Collections models."""
+
+from invenio_communities.communities.records.models import CommunityMetadata
+from invenio_db import db
+from invenio_records.models import Timestamp
+from sqlalchemy import UniqueConstraint
+from sqlalchemy_utils.types import UUIDType
+
+
+# CollectionTree Table
+class CollectionTree(db.Model, Timestamp):
+    """Collection tree model."""
+
+    __tablename__ = "collections_collection_tree"
+
+    __table_args__ = (
+        # Unique constraint on slug and community_id. Slugs should be unique within a community.
+        UniqueConstraint(
+            "slug",
+            "community_id",
+            name="uq_collections_collection_tree_slug_community_id",
+        ),
+    )
+
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    community_id = db.Column(
+        UUIDType,
+        db.ForeignKey(CommunityMetadata.id, ondelete="SET NULL"),
```

Comment:
```suggestion
        db.ForeignKey(CommunityMetadata.id, ondelete="RESTRICT"),
```

I think we got this one wrong in the initial model. We're not very consistent around table FKs with this, but I would say a good rule of thumb would be:

- If it's a "public" resource, we `RESTRICT`, so that you first have to consciously clean-up and make a decision on it. Community collections are "public" since one might expect links to them to work permanently (or redirect possibly if deleted).
- If it's a resource that is more like a paper-trail (e.g. a request responsible for a community member being added), we `SET NULL`.
- If it's none of the above, we `CASCADE`